google/cloud/spanner_v1/session.py

```diff
@@ -292,7 +296,9 @@ def transaction(self):
             self._transaction.rolled_back = True
             del self._transaction
```

Comment:
Not related to this particular PR, but this line may be redundant as `self._transaction` gets reassigned anyway.